lib-ng/scanners/base.js

```diff
@@ -121,13 +121,28 @@ var BaseScanner = function (beaconMapper, beaconRegionMapper, beaconRegionMonito
         const pathsenseLib = require('./../scanners/pathsense');
         pathsenseLib.init();
         pathsenseLib.stopMonitoring();
-        geofenceBeacons.forEach((beacon) => {
+        const geofenceRegions = geofenceBeacons.map((beacon) => {
             const identifier = `${beacon.get('beacon_id')}|${beacon.get('UUID')}|${beacon.get('major')}|${beacon.get('minor')}`;
-            pathsenseLib.startMonitoring({
+            return {
                 identifier,
                 latitude: beacon.get('latitude'),
                 longitude: beacon.get('longitude'),
                 radius: 100,
+            };
+        });
+        Ti.Geolocation.getCurrentPosition((e) => {
+            let nearestGeofenceRegions = geofenceRegions;
+            if (!e.success) {
+                nearestGeofenceRegions = geofenceRegions;
+            } else {
+                nearestGeofenceRegions = pathsenseLib.sortRegionsByDistance(geofenceRegions, {
+                    latitude: e.coords.latitude,
+                    longitude: e.coords.longitude,
+                });
+            }
+
+            nearestGeofenceRegions.forEach((region) => {
```

Comment:
Should we limit the amount to some constant like `MAX_REGIONS_MONITORED` which is 20 on iOS?